src/types.rs

```diff
@@ -23,42 +19,54 @@ impl Clone for Box<dyn Interface> {
     }
 }
 
+macro_rules! unimplemented {
```

Comment:
this shadows https://doc.rust-lang.org/std/macro.unimplemented.html

Reply:
That's the behavior wanted 😄 but I can rename it otherwise!